sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3545,6 +3561,44 @@ pi_result piextMemCreateWithNativeHandle(pi_native_handle NativeHandle,
   return PI_SUCCESS;
 }
 
+pi_result piextMemImgCreateWithNativeHandle(pi_native_handle NativeHandle,
+                                            pi_context Context,
+                                            bool OwnNativeHandle,
+                                            const pi_image_format *ImageFormat,
+                                            const pi_image_desc *ImageDesc,
+                                            pi_mem *RetImage) {
+
+  PI_ASSERT(RetImage, PI_ERROR_INVALID_VALUE);
+  PI_ASSERT(NativeHandle, PI_ERROR_INVALID_VALUE);
+  PI_ASSERT(Context, PI_ERROR_INVALID_CONTEXT);
+
+  std::shared_lock<pi_shared_mutex> Lock(Context->Mutex);
+
+  ze_image_handle_t ZeHImage = pi_cast<ze_image_handle_t>(NativeHandle);
+
+  try {
+    auto ZePIImage = new _pi_image(Context, ZeHImage, OwnNativeHandle);
+    *RetImage = ZePIImage;
+
+#ifndef NDEBUG
+    ZeStruct<ze_image_desc_t> ZeImageDesc;
+    pi_result DescriptionResult =
+        PIToZeImageDesc(ImageFormat, ImageDesc, ZeImageDesc);
+    if (DescriptionResult != PI_SUCCESS)
+      return DescriptionResult;
+
+    ZePIImage->ZeImageDesc = ZeImageDesc;
+#endif // !NDEBUG
```

Comment:
I don't know all the details but since this is only used in DEBUG should we event need to have `ImageDesc` as an argument to `piextMemImgCreateWithNativeHandle`?

Reply:
We definitely need it in the case of DEBUG though. `getImageRegionHelper`  and `enqueueMemImageCommandHelper`  just assume its existence. If the user is using DEBUG, we should have it. 

But they aren't a debug-only dependency.  The `make_image()` interface takes the sycl::image_channel_order , type and range which it needs to set up the `image_impl`.   Until LevelZero adds something like the equivalent of `clGetImageInfo` we will need those arguments to be able to realize `make_image`.

Reply:
> We definitely need it in the case of DEBUG though. `getImageRegionHelper` and `enqueueMemImageCommandHelper` just assume its existence. If the user is using DEBUG, we should have it.
> 
> But they aren't a debug-only dependency. The `make_image()` interface takes the sycl::image_channel_order , type and range which it needs to set up the `image_impl`. Until LevelZero adds something like the equivalent of `clGetImageInfo` we will need those arguments to be able to realize `make_image`.

Can you point to where it is used it out of DEBUG? (sounds strange since it is only declared in DEBUG)

Reply:
The `ImageDesc` is not used outside debug, but, in the bigger picture, the information that we construct the `ImageDesc` from ( image_channal_order, type, dimensions, etc), that information is needed to set up the `image_impl`. So, while we could arguable drop the ImageDesc from `piextMemImageCreateWithNativeHandle` we cannot change the similar arguments to `make_image`  .

And, if we remove the ImageDesc from `piextMemImageCreateWithNativeHandle` we'll also need to modify `getImageRegionHelper` and `enqueueMemImageCommandHelper` so they don't crash when in DEBUG mode. And, lastly, that would mean those helper functions would behave one way for an image created with `pieMemImageCreate` and another for an image created with `pieMemImageCreateWithNativeHandle`.  So if a user were trying to DEBUG, that would just be bewildering. 

IMO, we should keep it as written. If we want to get rid of the `ImageDesc`, we need  the LevelZero API to provide something like `clGetImageInfo` , but right now there is no such API.  If we had that, we could drop the extra info requirements from both `make_image` and `pieMemImageCreateWithNativeHandle`

Reply:
I agree, let's keep it as is.